### ISSUE TITLE
Handle websites with cookies better

### ISSUE THREAD

**Issue by Aerics84** [MohamedBassem/hoarder-app#50]
Is it possible to handle websites with an overlay for cookies better?
Many websites i add is shows no thumbnail and shows as the text only the cookies informations.
![grafik](https://github.com/MohamedBassem/hoarder-app/assets/58844333/0748aabb-9f57-4962-8fe8-24bf17b8163c)

**Comment by MohamedBassem** [comment]
Yeah, that's one area I know needs some improvement. I'm using a puppeteer that's supposed to be able to handle those banners, but it's not doing a great job (or I'm hindering it with some of the pre-processing). That's an improvement area indeed.

**Comment by mikkelnl** [comment]
I have the same 'issue' ;-) Subscribing to this topic to follow :-)

**Comment by lippoliv** [comment]
👍 

Google Maps, at least in Germany, ist Blocked by this as well:/

![Screenshot_20240802-115643](https://github.com/user-attachments/assets/c96cc9bb-8f22-4dcd-9297-c33457921716)

**Comment by ristomatti** [comment]
Could something like this approach work: https://stackoverflow.com/questions/59618456/pupeteer-how-can-i-accept-cookie-consent-prompts-automatically-for-any-url?

One comment on that answer included this example:

```js
/*
* Install some "no-cookies extension to your Chrome, 
* then copy the extension directory to your Pupeteer dir
*/

const puppeteer = require('puppeteer');
const fs = require('fs/promises');

async function start () {
    const pathToExtension = require('path').join(__dirname, 'extensions/3.4.5_0');
    const url = 'https://www.example.com/';
    const browser = await puppeteer.launch({
        headless: false,
        args: [
        `--disable-extensions-except=${pathToExtension}`,
        `--load-extension=${pathToExtension}`,
        ],
    });
    
    const page = await browser.newPage();
   
    await page.goto(url, {
        waitUntil: 'networkidle0',
      });

   
    await page.screenshot({path: 'screen1.png'});
    
    
    await browser.close();
}

start();
```

**Source:** https://gist.github.com/jirivrany/fa16f25d25142a074ab3ca14fe987eee. 

I do not know which extension the code comment refers to, but I found this fork of the popular "I don't care about cookies" extension that appears to be licensed under GPL 3.0: https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies.

**Comment by kriskras69** [comment]
I use "Bypass Paywall Clean" to get around that". It creates a copy of the website skipping the cookie part and then i hoard that page. But i would love to get extensions working headless.

**Comment by MohamedBassem** [comment]
let's track that in #414.
